src/plots/cartesian/select.js

```diff
@@ -308,18 +276,24 @@ function prepSelect(e, startX, startY, dragOptions, mode) {
 
         throttle.done(throttleID).then(function() {
             throttle.clear(throttleID);
+
             if(numClicks === 2) {
-                // clear selection on doubleclick
-                outlines.remove();
                 for(i = 0; i < searchTraces.length; i++) {
                     searchInfo = searchTraces[i];
-                    searchInfo._module.selectPoints(searchInfo, false);
+                    searchInfo._module.toggleSelected(searchInfo, false);
                 }
 
                 updateSelectedState(gd, searchTraces);
+
+                // clear visual boundaries of selection area if displayed
+                outlines.remove();
+
                 gd.emit('plotly_deselect', null);
             }
             else {
+                // TODO What to do with the code below because we now have behavior for a single click
```

Comment:
Thanks for writing down that `TODO`.

I'm thinking when `clickmode: 'select'` we'll emit `plotly_selected` with the selected points info. Otherwise, we'll have to keep that dummy event below.

Reply:
Thx.